pmlc/target/intel_gen_ocl_spirv/spirv_target.cc

```diff
@@ -15,12 +15,20 @@ namespace {
 class IntelGenOclAddSpirvTarget
     : public IntelGenOclAddSpirvTargetBase<IntelGenOclAddSpirvTarget> {
 public:
+  IntelGenOclAddSpirvTarget() = default;
+  explicit IntelGenOclAddSpirvTarget(unsigned spirvVersion) {
+    this->spirvVersion = spirvVersion;
+  }
   void runOnOperation() {
     auto target_env = getOperation().getAttrOfType<spirv::TargetEnvAttr>(
         spirv::getTargetEnvAttrName());
     if (!target_env) {
+      auto version = spirv::Version::V_1_5;
+      if (spirvVersion == 120) {
```

Comment:
Do we want to fix on 1.5 and 1.2 only or make it more generic?

Reply:
I feel like this is OK.  The `auto version = ` can signify the default (latest) version and we can add any other versions needed, currently just 1_2, in the conditional statements below.  Not sure how to make it more general, either.  I don't think there's a way to pass `spriv::Version` through the PassOptions directly.  Meaning, we have to translate from int (or string) to `spriv::Version` at some level.  Meaning, we need this case logic at some level.